scm/src/gmtb_scm_time_integration.F90

```diff
@@ -113,13 +117,42 @@ subroutine do_time_step(scm_state, physics, cdata, in_spinup)
       endif
     endif
   end do
-
+  
```

Comment:
Seeing that the same code exists twice, once in `scm/src/gmtb_scm_time_integration.F90` and once in `scm/src/gmtb_scm.F90`. Seems there is room for furthere optimization by putting all duplicate code in a separate file/routine and call it in both places?

Reply:
I have mixed thoughts on this. Yes it is repeated. However, the part that is repeated in gmtb_scm.F90 is only exercised when using the leapfrog scheme for the forcing terms for since the first timestep since it must be a forward timestep. I am likely going to remove the option to use a different time-stepping scheme due to the KISS principle and I don't think it is necessary to keep it. Once that is removed, the repetition will be gone.

Reply:
https://github.com/NCAR/ccpp-scm/issues/234